torch_xla/csrc/runtime/pjrt_registry.cc

```diff
@@ -56,13 +66,37 @@ InitializePjRt(const std::string& device_type) {
   std::unique_ptr<XlaCoordinator> coordinator;
 
   if (sys_util::GetEnvBool(env::kEnvPjrtDynamicPlugins, false)) {
-    std::optional<std::string> plugin_path = GetPjRtPluginPath(device_type);
-    if (plugin_path) {
+    std::optional<PluginEntry> plugin = GetPjRtPlugin(device_type);
+    if (plugin) {
       TF_VLOG(1) << "Initializing client for PjRt plugin " << device_type;
+
+      std::shared_ptr<xla::KeyValueStoreInterface> kv_store = nullptr;
+      if (plugin->init_coordinator) {
+        int global_process_rank = sys_util::GetEnvInt("RANK", 0);
+        int global_world_size = sys_util::GetEnvInt("WORLD_SIZE", 1);
```

Comment:
Sorry I didn't catch it earlier. What if the users start the single-host training in a non-torchrun way such as `PJRT_DEVICE=CUDA GPU_NUM_DEVICES=4 python3 xla/test/test_train_mp_imagenet.py`, then `global_world_size` should default to `local_world_size`? Similar reason for `global_process_rank`

Reply:
This case is slightly wrong. The precedence should be `$PJRT_LOCAL_PROCESS_COUNT`, then `$WORLD_SIZE`, then `1` (default). I'll fix it.

Reply:
Shouldn't it be WORLD_SIZE -> PJRT_LOCAL_PROCESS_COUNT or LOCAL_WORLD_SIZE -> 1?
If we use torchrun, then WORLD_SIZE will be set.
If we `GPU_NUM_DEVICES=4 python3 `, non-torchrun for single-host-multi-process, then WORLD_SIZE is not set and we rely on `PJRT_LOCAL_PROCESS_COUNT or LOCAL_WORLD_SIZE`.